math/mathcore/inc/Math/RanluxppEngine.h

```diff
@@ -0,0 +1,61 @@
+// @(#)root/mathcore:$Id$
+// Author: Jonas Hahnfeld 11/2020
+
+/*************************************************************************
+ * Copyright (C) 1995-2020, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_Math_RanluxppEngine
+#define ROOT_Math_RanluxppEngine
+
+#include "Math/TRandomEngine.h"
+
+#include <cstdint>
+#include <memory>
+
+namespace ROOT {
+namespace Math {
+
+template <int w, int p>
+class RanluxppEngineImpl;
+
+template <int p>
+class RanluxppEngine : public TRandomEngine {
```

Comment:
Does this class make sense to be derived from? If not maybe add final to hint the compiler to devirtualize the virtual methods.